tlair/src/test/scala/at/forsyte/apalache/tla/lir/io/TestCounterexampleWriter.scala

```diff
@@ -0,0 +1,219 @@
+        |
+        |State2 == x = 1 /\ y = 9
+        |
+        |(* Transition Trans2 to State3 *)
+        |
+        |State3 == x = 2 /\ y = 10
+        |
+        |(* The following formula holds true in the last state and violates the invariant *)
+        |
+        |InvariantViolation == x > 1 /\ y = 10
+        |
+        |================================================================================
+        |\* Created by Apalache on DATETIME
+        |\* https://github.com/konnov/apalache
+        |""".stripMargin
+    )
+  }
+
+  test("TLC single state") {
+    compare("tlc",
+      new TlaModule("test",List()),
+      gt(name("x"), int(1)),
+      List(
+        ("",Map("x" -> int(2)))
+      ),
+      """@!@!@STARTMSG 2262:0 @!@!@
+        |Created by Apalache on DATETIME
+        |@!@!@ENDMSG 2262 @!@!@
+        |@!@!@STARTMSG 2110:1 @!@!@
+        |Invariant is violated.
+        |@!@!@ENDMSG 2110 @!@!@
+        |@!@!@STARTMSG 2121:1 @!@!@
+        |The behavior up to this point is:
+        |@!@!@ENDMSG 2121 @!@!@
+        |@!@!@STARTMSG 2217:4 @!@!@
+        |1: <Initial predicate>
+        |x = 2
+        |
+        |@!@!@ENDMSG 2217 @!@!@
+        |""".stripMargin
+    )
+  }
+
+  test("TLC two steps") {
+    compare("tlc",
+      new TlaModule("test",List()),
+      gt(name("x"), int(1)),
+      List(
+        ("", Map("x" -> int(0))),
+        ("Next", Map("x" -> int(1))),
+        ("Next", Map("x" -> int(2)))
+      ),
+      """@!@!@STARTMSG 2262:0 @!@!@
+        |Created by Apalache on DATETIME
+        |@!@!@ENDMSG 2262 @!@!@
+        |@!@!@STARTMSG 2110:1 @!@!@
+        |Invariant is violated.
+        |@!@!@ENDMSG 2110 @!@!@
+        |@!@!@STARTMSG 2121:1 @!@!@
+        |The behavior up to this point is:
+        |@!@!@ENDMSG 2121 @!@!@
+        |@!@!@STARTMSG 2217:4 @!@!@
+        |1: <Initial predicate>
+        |x = 0
+        |
+        |@!@!@ENDMSG 2217 @!@!@
+        |@!@!@STARTMSG 2217:4 @!@!@
+        |2: <Next>
+        |x = 1
+        |
+        |@!@!@ENDMSG 2217 @!@!@
+        |@!@!@STARTMSG 2217:4 @!@!@
+        |3: <Next>
+        |x = 2
+        |
+        |@!@!@ENDMSG 2217 @!@!@
+        |""".stripMargin
+    )
+  }
+
+  test("TLC two steps with conjunction") {
+    compare("tlc",
+      new TlaModule("test",List()),
+      and(gt(name("x"), int(1)), eql(name("y"), int(10))),
+      List(
+        ("", Map("x" -> int(0), "y" -> int(8))),
+        ("Trans1", Map("x" -> int(1), "y" -> int(9))),
+        ("Trans2", Map("x" -> int(2), "y" -> int(10)))
+      ),
+      """@!@!@STARTMSG 2262:0 @!@!@
+        |Created by Apalache on DATETIME
+        |@!@!@ENDMSG 2262 @!@!@
+        |@!@!@STARTMSG 2110:1 @!@!@
+        |Invariant is violated.
+        |@!@!@ENDMSG 2110 @!@!@
+        |@!@!@STARTMSG 2121:1 @!@!@
+        |The behavior up to this point is:
+        |@!@!@ENDMSG 2121 @!@!@
+        |@!@!@STARTMSG 2217:4 @!@!@
+        |1: <Initial predicate>
+        |x = 0 /\ y = 8
```

Comment:
Does not TLC print one variable assignment per line in the counterexample, e.g.,
```tla
/\ x = 0
/\ y = 8
```

Reply:
yes, it does, but it doesn't matter -- both forms are valid and accepted by TLA Toolbox, I've checked.

Reply:
It might be harder for third-party scripts to parse it though. Think of our work on testing.